File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitAssetWorkSheet/EngineerVisitAssetWorkSheetValidatorUnvented.vb
Namespace Entity

Namespace EngineerVisitAssetWorkSheets

        Public Class EngineerVisitAssetWorkSheetValidatorUnvented
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerVisitAssetWorkSheet As EngineerVisitAssetWorkSheet)

                'make sure that contact object is valid
                If oEngineerVisitAssetWorkSheet.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerVisitAssetWorkSheet.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oEngineerVisitAssetWorkSheet.AssetID = Nothing Then
                    Me.AddCriticalMessage("Appliance Safe Missing")
                End If

                If oEngineerVisitAssetWorkSheet.LandlordsApplianceID = Nothing Then
                    Me.AddCriticalMessage("Water pressure Missing")
                End If

                If oEngineerVisitAssetWorkSheet.FlueFlowTestID = Nothing Then
                    Me.AddCriticalMessage("Cylinder integrity Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SpillageTestID = Nothing Then
                    Me.AddCriticalMessage("Pre-Charge pressure Missing")
                End If

                If oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Filter/Strainer check  Missing")
                End If

                If oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Sacrificial anode check Missing")
                End If

                If oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID = Nothing Then
                    Me.AddCriticalMessage("Expansion Gap check  Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID = Nothing Then
                    Me.AddCriticalMessage("Tundish check Missing")
                End If

                If oEngineerVisitAssetWorkSheet.SweepOutcomeID = Nothing Then
                    Me.AddCriticalMessage("Connection check Missing")
                End If

                If oEngineerVisitAssetWorkSheet.OverallID = Nothing Then
                    Me.AddCriticalMessage("Discharge pipe checks Missing")
                End If

                If oEngineerVisitAssetWorkSheet.DischargeID = Nothing Then
                    Me.AddCriticalMessage("temp/ pressure relief value Missing")
                End If

                If oEngineerVisitAssetWorkSheet.TankID = Nothing Then
                    Me.AddCriticalMessage("expansion value check Missing")
                End If

                If oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID = Nothing Then
                    Me.AddCriticalMessage("water pressure / flow rates check  Missing")
                End If

                If oEngineerVisitAssetWorkSheet.ApplianceSafeID = Nothing Then
                    Me.AddCriticalMessage("Service record  Missing")
                End If


                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitAssetWorkSheet/EngineerVisitAssetWorkSheetValidatorUnvented.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVisitAssetWorkSheets
    {
        // make sure that contact object is valid
        public class EngineerVisitAssetWorkSheetValidatorUnvented : BaseValidator
        {
            public void Validate(EngineerVisitAssetWorkSheet oEngineerVisitAssetWorkSheet)
            {
                // make sure that contact object is valid
                if (oEngineerVisitAssetWorkSheet.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oEngineerVisitAssetWorkSheet.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oEngineerVisitAssetWorkSheet.AssetID == default)
                {
                    AddCriticalMessage("Appliance Safe Missing");
                }

                if (oEngineerVisitAssetWorkSheet.LandlordsApplianceID == default)
                {
                    AddCriticalMessage("Water pressure Missing");
                }

                if (oEngineerVisitAssetWorkSheet.FlueFlowTestID == default)
                {
                    AddCriticalMessage("Cylinder integrity Missing");
                }

                if (oEngineerVisitAssetWorkSheet.SpillageTestID == default)
                {
                    AddCriticalMessage("Pre-Charge pressure Missing");
                }

                if (oEngineerVisitAssetWorkSheet.FlueTerminationSatisfactoryID == default)
                {
                    AddCriticalMessage("Filter/Strainer check  Missing");
                }

                if (oEngineerVisitAssetWorkSheet.VisualConditionOfFlueSatisfactoryID == default)
                {
                    AddCriticalMessage("Sacrificial anode check Missing");
                }

                if (oEngineerVisitAssetWorkSheet.VentilationProvisionSatisfactoryID == default)
                {
                    AddCriticalMessage("Expansion Gap check  Missing");
                }

                if (oEngineerVisitAssetWorkSheet.SafetyDeviceOperationID == default)
                {
                    AddCriticalMessage("Tundish check Missing");
                }

                if (oEngineerVisitAssetWorkSheet.SweepOutcomeID == default)
                {
                    AddCriticalMessage("Connection check Missing");
                }

                if (oEngineerVisitAssetWorkSheet.OverallID == default)
                {
                    AddCriticalMessage("Discharge pipe checks Missing");
                }

                if (oEngineerVisitAssetWorkSheet.DischargeID == default)
                {
                    AddCriticalMessage("temp/ pressure relief value Missing");
                }

                if (oEngineerVisitAssetWorkSheet.TankID == default)
                {
                    AddCriticalMessage("expansion value check Missing");
                }

                if (oEngineerVisitAssetWorkSheet.ApplianceServiceOrInspectedID == default)
                {
                    AddCriticalMessage("water pressure / flow rates check  Missing");
                }

                if (oEngineerVisitAssetWorkSheet.ApplianceSafeID == default)
                {
                    AddCriticalMessage("Service record  Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}